### DISCUSSION TITLE
Videos buffer on LAN and WAN

### DISCUSSION THREAD

**Discussion by poblabs** [immich-app/immich#3742] [Q&A]
Hey there, I am using a pretty vanilla docker-compose of immich and I'm noticing that videos - short and long - buffer every 1s when on LAN and WAN. 

I seem to notice the most on mobile devices on LAN and WAN

My server I'm running on is a dual Xeon E5-2680 with 48 cores, and 32gb of ram. Plenty of hardware so I think the issue may be Docker and not enough resource allocation?

Since there's 8 services in the docker-compose, which service should I allocate more CPU and RAM to so that the buffering can be reduced?

**Comment by mertalev** [comment]
This sounds more like a networking issue than a CPU one. Streaming isn't very CPU-intensive: even a Raspberry Pi 4 can keep up.

**Reply by poblabs** [reply]
I would agree except it's fine on PC. Not fine on mobile. I guess I left that detail out by mistake (I'll edit my original). 

So maybe it's an transcoding issue?

**Reply by mertalev** [reply]
It's possible if the videos aren't transcoded or the transcoding quality is so high that it blows up the file size. A few questions:

1. Have you changed any of the transcoding settings? If so, what are the changes?
2. Are all of your videos transcoded? If not, are the videos you're trying to play actually transcoded?

**Reply by poblabs** [reply]
1. I'm not sure - where would I change those settings? I have everything default right now

2. Not sure about this one either. I'm just uploading them. It works fine on PC, but not on mobile. So I assume Immich is transcoding these videos to be playable on mobile? (even though the videos came from my mobile)

**Reply by mertalev** [reply]
To verify that your videos are transcoded, you can run a "missing" job for transcoding in the Jobs page. But if your settings are default, I imagine they'd be done with a CPU that powerful. You might want to make an issue for this since it seems like a bug with mobile.

Edit: Also to clarify, videos are transcoded only once. The same transcoded video is streamed to all clients.

**Reply by poblabs** [reply]
Thanks. I just clicked the missing job for the transcoded videos and when it completed I clicked on a shared video link. Now it's not playing on PC anymore without buffering every 1s 🤔 

I'll open an issue

![image](https://github.com/immich-app/immich/assets/3484775/7d08e774-ff6c-4a77-9cd7-f09c2a22c950)

**Reply by poblabs** [reply]
https://github.com/immich-app/immich/issues/3766
